plugin/trino-hive/src/test/java/io/trino/plugin/hive/avro/TestAvroSchemaEvolutionWithPartitioning.java

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright Starburst Data, Inc. All rights reserved.
+ *
+ * THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF STARBURST DATA.
+ * The copyright notice above does not evidence any
+ * actual or intended publication of such source code.
+ *
+ * Redistribution of this material is strictly prohibited.
+ */
+package io.trino.plugin.hive.avro;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.containers.HiveMinioDataLake;
+import io.trino.plugin.hive.s3.S3HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.MaterializedResult.resultBuilder;
+import static java.lang.String.format;
+
+@Test
+public class TestAvroSchemaEvolutionWithPartitioning
+        extends AbstractTestQueryFramework
+{
+    private static final String BUCKET_NAME = "avro-bucket";
+
+    private static final String CATALOG_NAME = "hive";
+    private static final String SCHEMA_NAME = "avro";
+    private static final String TABLE_NAME = "avro_partitioning";
+
+    private static final String ORIGINAL_SCHEMA_FILE_NAME = "original_schema.avsc";
+    private static final String RENAME_COLUMN_SCHEMA_FILE_NAME = "rename_column_schema.avsc";
+    private static final String REMOVE_COLUMN_SCHEMA_FILE_NAME = "remove_column_schema.avsc";
+    private static final String ADD_COLUMN_SCHEMA_FILE_NAME = "add_column_schema.avsc";
+    private static final String CHANGE_COLUMN_TYPE_SCHEMA_FILE_NAME = "change_column_type_schema.avsc";
+    private static final String INCOMPATIBLE_TYPE_SCHEMA_FILE_NAME = "incompatible_type_schema.avsc";
+
+    private static final String SELECT_STAR_STATEMENT = prepareSelectStartStatement(TABLE_NAME);
```

Comment:
Better if we could test on queries with a specific columns.

Reply:
Ok will refactor test to pass specific column names